lib/qesapdeployment.pm

```diff
@@ -786,77 +786,92 @@ sub qesap_calculate_az_address_range {
+    record_info("PEERING SUCCESS (target)", "[M] Peering from $args{target_group}.$target_vnet server was successful\n");
 
     record_info("Checking peering status");
-    assert_script_run("az network vnet peering show --name $peering_name --resource-group $args{target_group} --vnet-name $args{target_vnet} --output table");
+    assert_script_run("az network vnet peering show --name $peering_name --resource-group $args{target_group} --vnet-name $target_vnet --output table");
     record_info("PEERING STATUS SUCCESS");
 }
 
 =head3 qesap_delete_az_peering
 
-    Performs peering between the cluster and IBS mirror on Azure.
+    Delete all the network peering between the two provided deploymnets.
 
-=over 2
+=over 3
 
-=item B<SOURCE_GROUP> - resource group of source
+=item B<SOURCE_GROUP> - resource group of source.
+                        This parameter is optional, if not provided
+                        the related peering will be ignored.
 
-=item B<TARGET_GROUP> - resource group of target
+=item B<TARGET_GROUP> - resource group of target.
+                        This parameter is mandatory and
+                        the associated resource group is supposed to still exist.
+
+=item B<TIMEOUT> - default is 5 mins
 
 =back
 =cut
 
 sub qesap_delete_az_peering {
     my (%args) = @_;
     croak 'Missing mandatory target_group argument' unless $args{target_group};
+    $args{timeout} //= bmwqemu::scale_timeout(300);
 
-    $args{target_vnet} = qesap_get_vnet($args{target_group});
+    my $target_vnet = qesap_get_vnet($args{target_group});
 
-    my $peering_name = qesap_get_peering_name();
+    my $peering_name = qesap_get_peering_name($args{target_group});
     if (!$peering_name) {
-        record_info("NO PEERING", "No peering to be destroyed!");
+        record_info('NO PEERING', "No peering between $args{target_group} and resources belonging to the current job to be destroyed!");
         return;
     }
 
     record_info("Attempting peering destruction");
+    my $peering_cmd = "az network vnet peering delete -n $peering_name";
     my $source_ret = 0;
     if ($args{source_group}) {
         $args{source_vnet} = qesap_get_vnet($args{source_group});
-        my $source_cmd = "az network vnet peering delete --resource-group $args{source_group} --vnet-name $args{source_vnet} -n $peering_name";
-        $source_ret = script_run($source_cmd);
+        my $source_cmd = "$peering_cmd --resource-group $args{source_group} --vnet-name $args{source_vnet}";
+        $source_ret = script_run($source_cmd, timeout => $args{timeout});
     }
-    my $target_cmd = "az network vnet peering delete --resource-group $args{target_group} --vnet-name $args{target_vnet} -n $peering_name";
-    my $target_ret = script_run($target_cmd);
+    my $target_cmd = "$peering_cmd --resource-group $args{target_group} --vnet-name $args{target_vnet}";
+    my $target_ret = script_run($target_cmd, timeout => $args{timeout});
```

Comment:
You can also do:
```
$source_ret = script_run($source_cmd, %args);
```
And that way you can pass other arguments in `%args` besides only the timeout.